### PR TITLE
test getInvokeEstimateFee

### PR DESCRIPTION
**Pull request by financegeek-org**
## Motivation and Resolution
Requested in https://github.com/0xs34n/starknet.js/issues/589

...

### RPC version (if applicable)

...

## Usage related changes
N/A

## Development related changes
N/A; SDK test

## Checklist:

- [X] Performed a self-review of the code
- [X] Rebased to the last commit of the target branch (or merged it into my branch)
- [X] Linked the issues which this PR resolves
- [ ] Documented the changes in code (API docs will be generated automatically)
- [X] Updated the tests
- [ ] All tests are passing

### CODE REVIEW

**Review by ivpavici — APPROVED** [review]
lgtm

File: __tests__/rpcProvider.test.ts [hunk 1]
```diff
@@ -99,6 +105,29 @@ describeIfRpc('RPCProvider', () => {
...
+    test('getInvokeEstimateFee', async () => {
+      const fetchSpy = jest.spyOn(rpcProvider as any, 'fetchEndpoint');
+
+      const randomWallet = stark.randomAddress();
+      const invocation: Invocation = {
+        contractAddress: randomWallet,
+        entrypoint: 'transfer',
+        calldata: [randomWallet, '10', '0'],
+        signature: [],
+      }
```
Comment (tabaktoni) [inline __tests__/rpcProvider.test.ts:117]: Multiple linting issues

```suggestion
      };
```

File: __tests__/rpcProvider.test.ts [hunk 2]
```diff
@@ -99,6 +105,29 @@ describeIfRpc('RPCProvider', () => {
...
+        nonce: '',
+      }
+      const transaction = await rpcProvider.getInvokeEstimateFee(
+        invocation,
+        invocationDetails,
+        latestBlock.block_number,
+      );
+      expect(fetchSpy).toHaveBeenCalledWith('starknet_estimateFee', expect.anything());
+    });
+
```
Comment (tabaktoni) [inline __tests__/rpcProvider.test.ts:129]:
```suggestion
```

File: __tests__/rpcProvider.test.ts [hunk 3]
```diff
@@ -99,6 +105,29 @@ describeIfRpc('RPCProvider', () => {
...
+      const invocation: Invocation = {
+        contractAddress: randomWallet,
+        entrypoint: 'transfer',
+        calldata: [randomWallet, '10', '0'],
+        signature: [],
+      }
+      const invocationDetails: InvocationsDetailsWithNonce = {
+        ...invocation,
+        nonce: '',
+      }
```
Comment (tabaktoni) [inline __tests__/rpcProvider.test.ts:121]:
```suggestion
      };
```

File: __tests__/rpcProvider.test.ts [hunk 4]
```diff
@@ -99,6 +105,29 @@ describeIfRpc('RPCProvider', () => {
...
+      const transaction = await rpcProvider.getInvokeEstimateFee(
```
Comment (tabaktoni) [inline __tests__/rpcProvider.test.ts:122]: transaction var is not used
```suggestion
      await rpcProvider.getInvokeEstimateFee(
```

**Review by tabaktoni — CHANGES_REQUESTED** [review]
1. Testing this is redundant but if we test it as unit then it should be tested for both providers RPC and Sequencer.
2. Multiple linting issues fixed suggested
3. LibraryError: 20: Contract not found when test run: You can't ping random address
4. You are testing that method is called but you called it with `rpcProvider.getInvokeEstimateFee` unnecessary test
5. You do not test the execution of this method nor its functional or execution response.

In short, nothing is correct in this PR.
I recommend setting it to draft until it is completed, or closing it and publishing fixed one
If you need help with improving comprehension ping us on Discord and we will be glad to assist with it.

**Comment by financegeek-org** [comment]
> 2. Multiple linting issues fixed suggested
Updated, thanks!

> 3. LibraryError: 20: Contract not found when test run: You can't ping random address
So create a contract and deploy, then test that? 🤔 

> 4. You are testing that method is called but you called it with `rpcProvider.getInvokeEstimateFee` unnecessary test
It's testing getInvokeEstimateFee by checking that fetchEndpoint is called with parameter starknet_estimateFee.

> 5. You do not test the execution of this method nor its functional or execution response.
Suggestions on how to test this?

**Comment by penovicp** [comment]
Closing this because the underlying code has undergone significant changes. Ask in the corresponding issue if you want to make another attempt.
